onboarding/src/main/kotlin/io/homeassistant/companion/android/onboarding/nameyourdevice/NameYourDeviceScreen.kt

```diff
@@ -0,0 +1,211 @@
+            contentDescription = null,
+        )
+        Text(
+            text = stringResource(R.string.name_your_device_title),
+            style = HATextStyle.Headline,
+        )
+        Text(
+            text = stringResource(R.string.name_your_device_content),
+            style = HATextStyle.Body,
+        )
+
+        DeviceNameTextField(
+            deviceName = deviceName,
+            onDeviceNameChange = onDeviceNameChange,
+            saveClickable = saveClickable,
+            onSaveClick = onSaveClick,
+            deviceNameEditable = deviceNameEditable,
+        )
+
+        Spacer(modifier = Modifier.weight(1f))
+
+        HAAccentButton(
+            text = stringResource(R.string.name_your_device_save),
+            onClick = onSaveClick,
+            enabled = saveClickable,
+            modifier = Modifier
+                .padding(bottom = HASpacing.XL),
+        )
+    }
+}
+
+@Composable
+private fun DeviceNameTextField(
+    deviceName: String,
+    onDeviceNameChange: (String) -> Unit,
+    saveClickable: Boolean,
+    deviceNameEditable: Boolean,
+    onSaveClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    val focusRequester = remember { FocusRequester() }
+
+    HATextField(
+        value = deviceName,
+        onValueChange = onDeviceNameChange,
+        trailingIcon = {
+            if (deviceName.isNotEmpty()) {
+                IconButton(onClick = { onDeviceNameChange("") }, enabled = deviceNameEditable) {
+                    Icon(
+                        imageVector = Icons.Default.Close,
+                        contentDescription = stringResource(R.string.name_your_device_clear_name),
+                    )
+                }
+            }
+        },
+        keyboardOptions = KeyboardOptions(imeAction = ImeAction.Done),
+        keyboardActions = KeyboardActions(
+            onDone = {
+                focusRequester.freeFocus()
+                if (saveClickable) {
+                    onSaveClick()
+                }
+                // This is going to hide the keyboard and clear focus on the text field
+                defaultKeyboardAction(ImeAction.Done)
+            },
+        ),
+        enabled = deviceNameEditable,
+        modifier = modifier
+            .focusRequester(focusRequester)
+            .testTag(DEVICE_NAME_TEXT_FIELD_TAG),
+    )
+    // Request focus on the text field when the screen is shown
+    LaunchedEffect(Unit) {
+        focusRequester.requestFocus()
+    }
+}
+
+@HAPreviews
+@Composable
+private fun NameYourDeviceScreenPreview() {
+    HAThemeForPreview {
+        NameYourDeviceScreen(
+            onHelpClick = {},
+            onBackClick = {},
+            deviceName = "Superman",
+            onDeviceNameChange = {},
+            saveClickable = true,
+            deviceNameEditable = true,
+            onSaveClick = {},
+        )
+    }
+}
+
+// TODO check that we call device registration with the name set
+// TODO write all the tests
```

Comment:
- You are doing that
- I'm not sure if it's all you wanted, but I'd remove it or be more specific